### DISCUSSION TITLE
additional benefits of this approach

### DISCUSSION THREAD

**Discussion by rdmueller** [diagramsascode/diagramsascode#4] [Ideas]
I just wonder if it would be possible to use the underlying code for the diagrams as model?

Imagine you define all your nodes and edges in several files as model and then select which part or view of the model to be displayed?

I think the Structurizr approach gets already close to it.

**Comment by diagramsascode** [comment]
Can you provide an example, so that I can understand it better?

**Reply by rdmueller** [reply]
let's see...

if I define a model with nodes and connections (in a fantasy DSL) like this:

```
component A with name "Alice"
component B with name "Bob"
component C with name "Charly"
B uses A
B inherits from C
```

then it would be great to define a view/diagram as

```
drawContext(A); // only shows A and B
drawContext(B); // shows A, B and C
```

and a change in the model would be reflected in all views: if I rename "Alice" with "Alex", all views could be easily updated.

Now that I compare this with your example from the readme, I think it is already pretty close to it. :-D

**Reply by diagramsascode** [reply]
Yes, this can be realized by having several `Diagram` builders, and passing in a shared subset of the nodes & edges, and elements special to the diagram.
